### DISCUSSION TITLE
canvas vs offscreencanvas (performance enhancement ?)

### DISCUSSION THREAD

**Discussion by eviltik** [felixmariotto/three-mesh-ui#174] [Ideas]
When using a canvas, the main browser thread is used. 

More Blocks, more mesh, more canvas, . I noticed that the keyboard example is a little (really little) bit lagging on my oculus quest browser.

When using an offscreencanvas, a dedicated worker is used (per offscreencanvas instance). It's suppose to be faster, let the main thread do his job rather than handling canvas.

Offscreencanvas is not available for all browsers yet : https://caniuse.com/offscreencanvas

So what i'm testing is:
* make canvas generation a class
    * check if offscreencanvas is available, if yes use it, else fallback native canvas.
* see if "lags" always here using offscreencanvas
* compare fps

I'll give feedbacks on this.

**Comment by eviltik** [comment]
My bad, i didn't take enought time to understand that three-mesh-ui use shaders rather than canvas. This discussion does not make sense. 

I don't see any easy way to improve performance related to the keyboard example.

**Reply by swingingtom** [reply]
Thank anyway for looking through that door
